packages/nifti-volume-loader/src/helpers/fetchAndAllocateNiftiVolume.ts

```diff
@@ -0,0 +1,263 @@
+import * as NiftiReader from 'nifti-reader-js';
+import {
+  cache,
+  utilities,
+  Enums,
+  eventTarget,
+  triggerEvent,
+} from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import { vec3 } from 'gl-matrix';
+import makeVolumeMetadata from './makeVolumeMetadata';
+import NiftiImageVolume from '../NiftiImageVolume';
+import * as NIFTICONSTANTS from './niftiConstants';
+import { invertDataPerFrame, rasToLps } from './convert';
+import scaleNiftiArray from './scaleNiftiArray';
+import Events from '../enums/Events';
+
+const { createUint8SharedArray, createFloat32SharedArray } = utilities;
+
+export const urlsMap = new Map();
+
+function fetchArrayBuffer(url, onProgress, signal, onload) {
+  return new Promise((resolve, reject) => {
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', url, true);
+
+    xhr.responseType = 'arraybuffer';
+
+    xhr.onload = function (e) {
+      if (onload && typeof onload === 'function') {
+        onload();
+      }
+
+      resolve(xhr.response);
+    };
+
+    if (onProgress && typeof onProgress === 'function') {
+      xhr.onprogress = function (e) {
+        //  if (e.lengthComputable) {
+        //    onProgress(e.loaded, e.total);
+        //  }
+        onProgress(e.loaded, e.total);
+      };
+    }
+
+    if (signal && signal.aborted) {
+      xhr.abort();
+      reject(new Error('Request aborted'));
+    }
+
+    signal.addEventListener('abort', () => {
+      xhr.abort();
+      reject(new Error('Request aborted'));
+    });
+
+    xhr.send();
+  });
+}
+
+export const getTypedNiftiArray = (datatypeCode, niftiImageBuffer) => {
+  switch (datatypeCode) {
+    case NIFTICONSTANTS.NIFTI_TYPE_UINT8:
+      return new Uint8Array(niftiImageBuffer);
+    case NIFTICONSTANTS.NIFTI_TYPE_FLOAT32:
+      return new Float32Array(niftiImageBuffer);
+    case NIFTICONSTANTS.NIFTI_TYPE_INT16:
+      return new Int16Array(niftiImageBuffer);
+    default:
+      throw new Error(`datatypeCode ${datatypeCode} is not yet supported`);
+  }
+};
+
+export default async function fetchAndAllocateNiftiVolume(
+  volumeId: string
+): Promise<NiftiImageVolume> {
+  // nifti volumeIds start with 'nifti:' so we need to remove that
```

Comment:
Just a suggestion, but does it make sense to declare a constant somewhere called (for example) `nifitiFilePrefix = 'nifti:'` and then the following line would be `volumeId.substring(niftiFilePrefix.length)`?